protocol/packages/swap/tests/request.rs

```diff
@@ -0,0 +1,60 @@
+use currency::{
+    test::{SubGroup, SubGroupTestC1, SuperGroup, SuperGroupTestC2, SuperGroupTestC3},
+    Currency as _,
+};
+use dex::swap::{ExactAmountIn, SwapRequest};
+use oracle::api::swap::SwapTarget;
+use platform::trx::Transaction;
+use sdk::{
+    cosmos_sdk_proto::Any as CosmosAny, cosmwasm_std::Binary,
+    neutron_sdk::bindings::types::ProtobufAny as NeutronAny,
+};
+use swap::Impl;
+
+#[test]
+fn build_and_parse() {
+    let expected_token_in =
+        finance::coin::from_amount_ticker::<SubGroup>(20, SubGroupTestC1::TICKER.into()).unwrap();
```

Comment:
we have got a much more preferred way of creating Coin<DTO>
`let expected_token_in: CoinDTO<SubGroup> = Coin::<SubGroupTestC1>::from(20).into()`